### PR TITLE
chore: Removing `examples` module

### PR DESCRIPTION
**Pull request by zepfred**
This PR removes the `examples` module.

### CODE REVIEW

**Comment by triceo** [comment]
Depends on:
https://github.com/TimefoldAI/timefold-solver/pull/800

Let's also add a note to the upgrade recipe:
https://docs.timefold.ai/timefold-solver/latest/upgrade-and-migration/upgrade-to-latest-version#_upgrade_from_1_8_0_to_1_9_0

Something along the lines that we've removed all of the legacy examples and turned them into quickstarts.

**Review by triceo — APPROVED** [review]
The predecessor PR has been merged.
This may be merged once rebased on latest `main`.

File: docs/src/modules/ROOT/pages/upgrade-and-migration/upgrade-to-latest-version.adoc [hunk 1]
```diff
@@ -140,6 +140,43 @@ public class Lesson {
 ----
 
 
+====
+
+'''
+
+=== Upgrade from 1.9.0 to 1.10.0
```
Comment (triceo) [inline docs/src/modules/ROOT/pages/upgrade-and-migration/upgrade-to-latest-version.adoc:147]: This doesn't belong here, it's already above.

**Comment by triceo** [comment]
I don't understand why, or why this PR would cause it, but the test is failing consistently. I re-ran it and it also failed.

**Comment by zepfred** [comment]
> I don't understand why, or why this PR would cause it, but the test is failing consistently. I re-ran it and it also failed.

It's strange because it's passing locally.

**Comment by zepfred** [comment]
@triceo It is now passing, but I am unsure what happened before. I just applied the necessary upgrades and promoted the PR to 'ready for review'.

**Review by triceo — APPROVED** [review]

**Pull request merged**